test/test.js

```diff
@@ -16,44 +16,54 @@ const testArray = [{
 }]
 
 describe('OatyArray', () => {
-  it('should initialise', () => {
-    const oat = new OatyArray(testArray, ['a', 'b', 'fruit'])
+  describe('Constructor', () => {
+    it('should initialise', () => {
+      const oat = new OatyArray({}) 
+      expect(oat).to.exist 
+    }) 
 
-    expect(oat).to.exist
-  })
+    it('should initialise with data', () => {
+      const oat = new OatyArray({data: testArray}) 
+      expect(oat).to.exist 
+    }) 
 
-  it('should match a search', () => {
-    const oat = new OatyArray(testArray, ['fruit'])
-    const matches = oat.get('fruit', 'banana')
+    it('should initialise with keys', () => {
+      const oat = new OatyArray({keys: ['a', 'b', 'fruit']}) 
+      expect(oat).to.exist 
+    }) 
 
-    expect(matches).to.not.be.undefined
-    expect(matches[0]).to.not.be.undefined
-    expect(matches[0].b).to.equal(3)
+    it('should initialise with data and keys', () => {
+      const oat = new OatyArray(testArray, {keys: ['a', 'b', 'fruit']}) 
+      expect(oat).to.exist 
+    }) 
   })
 
-  it('should let the user query pushed objects', () => {
-    const oat = new OatyArray(testArray, ['fruit'])
-
-    oat.push(
-      { a: 5, b: 5, fruit: 'potato' },
-      { a: 6, b: 6, fruit: 'courgette' }
-    )
-
-    const matches = oat.get('fruit', 'potato')
-
-    expect(matches).to.not.be.undefined
-    expect(matches[0]).to.not.be.undefined
-    expect(matches[0].b).to.equal(5)
+  describe('Push', () => {
+    it('push function should return count', () => {

```

Comment:
usually `describe` and `it` are meant to make sense in terms of grammar - maybe make it something like:

```ts
describe('push function', () => {
  it('should return count', () => {
```

which reads as "push function should return count", for example

Reply:
I was going to rename them all when we add more tests to follow "given when then" but I'll do it now

Reply:
In fact, I also need to add some more tests so I'll remove the data and use fixtures

Reply:
Outdated